update random tests

SmarterHomes opened the pull request.

d4ndo commented:
Please wait with this Pullrequest. /src/qt/Makefile needs modification.

d4ndo commented:
Modified the Makefile and replaced the »spinner«, »connection« and »tx« icons. Added a new label »logo« that can be set by stylesheet(theme).
P.S. A new theme is in work.

d4ndo commented:
The chaincoin community still needs to test the changes.

rolichain commented:
Thanks, the two sources you mentioned appear to be MIT license friendly, so that's good.

d4ndo commented:
I had to undo some stuff. The changes made did not compile as they should, because of some Qt-Plugins. I tried to move to svg icons, that also require the SVG Qt Plugin. So all we have won are the modern icon theme. 

Maybe you could provide another test client for the community. If no complains pop up we could continue.

Commit:
Provide relevant error message if datadir is not writable.

sipa committed:
Make FastRandomContext support standard C++11 RNG interface

This makes it possible to plug it into the various standard C++11 random
distribution algorithms and other functions like std::shuffle.
